src/types.rs

```diff
@@ -7,26 +7,55 @@ use std::{
     task::{Context, Poll},
 };
 
-use actix_http::body::BodySize;
 use actix_http::body::MessageBody;
+use actix_http::body::{BodySize, BoxBody};
+use actix_http::StatusCode;
 use actix_web::web::Bytes;
 use actix_web::{http::Method, HttpRequest};
-
+use actix_web::{HttpResponse, HttpResponseBuilder, Responder};
 use anyhow::Result;
 use dashmap::DashMap;
 use pyo3::exceptions::PyValueError;
 use pyo3::types::{PyBytes, PyString};
 use pyo3::{exceptions, prelude::*};
 
-use crate::io_helpers::read_file;
+use crate::io_helpers::{apply_hashmap_headers, read_file};
 
 fn type_of<T>(_: &T) -> String {
     std::any::type_name::<T>().to_string()
 }
 
-#[derive(Debug, Clone)]
-#[pyclass]
-pub struct ActixBytesWrapper(Bytes);
+#[derive(Debug, Clone, Default)]
+#[pyclass(name = "Body")]
+pub struct ActixBytesWrapper {
+    content: Bytes,
+}
+
+#[pymethods]
+impl ActixBytesWrapper {
+    pub fn as_str(&self) -> PyResult<String> {
+        Ok(String::from_utf8(self.content.to_vec())?)
+    }
+
+    pub fn as_bytes(&self) -> PyResult<Vec<u8>> {
+        Ok(self.content.to_vec())
+    }
+
+    pub fn set(&mut self, content: &PyAny) -> PyResult<()> {
+        let value = if let Ok(v) = content.downcast::<PyString>() {
+            v.to_string().into_bytes()
+        } else if let Ok(v) = content.downcast::<PyBytes>() {
+            v.as_bytes().to_vec()
+        } else {
+            return Err(PyValueError::new_err(format!(
+                "Could not convert {} specified body to bytes",
+                type_of(content)
+            )));
+        };
+        self.content = Bytes::from(value);
+        Ok(())
+    }
```

Comment:
I have a suggestion here. How about using the `#[getter]` and `#[setter]` here(https://pyo3.rs/main/class#object-properties-using-getter-and-setter)

We can keep the base content as `bytes` but then override the getters and setters to string. 

And then only keep an `as_bytes` method when we want to analyze the byte response.